iphone/Classes/TiUIListView.m

```diff
@@ -2225,22 +2192,22 @@ -(CGFloat)contentHeightForWidth:(CGFloat)width
 -(void)clearSearchController:(id)sender
 {
     if (sender == self) {
-        RELEASE_TO_NIL(tableController);
-        RELEASE_TO_NIL(searchController);
         [searchViewProxy ensureSearchBarHeirarchy];
     }
 }
 
 -(void)initSearchController:(id)sender
 {
-    if (sender == self && tableController == nil) {
-        tableController = [[UITableViewController alloc] init];
-        [TiUtils configureController:tableController withObject:nil];
-        tableController.tableView = [self tableView];
-        searchController = [[UISearchDisplayController alloc] initWithSearchBar:[searchViewProxy searchBar] contentsController:tableController];
-        searchController.searchResultsDataSource = self;
-        searchController.searchResultsDelegate = self;
+    if (sender == self && searchController == nil)
+    {
+        searchController = [[UISearchController alloc] initWithSearchResultsController:nil];
```

Comment:
In UISearchController there is option to pass nil for result controller. In this case it will use the current tableview for showing the search result. We can not use different controller to show the result because it will create similar problem which is mentioned in TIMOB-16622 and TIMOB-16826.